### PR TITLE
Add the dual fields to the paraview output

### PR DESCRIPTION
**Pull request by jamiebramwell**
Previously, only the `FiniteElementStates` (namely the primal solution fields and parameters) were output in Paraview format. This adds the dual fields (e.g. reactions and shape displacement sensitivities) as well for visualization.

### CODE REVIEW

**Comment by jamiebramwell** [comment]
@btalamini @white238 @barreracruz , could you take a look at this?

File: src/serac/physics/base_physics.cpp [hunk 1]
```diff
@@ -151,6 +151,18 @@ void BasePhysics::outputStateToDisk(std::optional<std::string> paraview_output_d
...
+        serac::FiniteElementDual* non_const_dual = const_cast<serac::FiniteElementDual*>(dual);
+
+        paraview_dual_grid_functions_[dual->name()] =
+            std::make_unique<mfem::ParGridFunction>(const_cast<mfem::ParFiniteElementSpace*>(&dual->space()));
+        non_const_dual->space().GetRestrictionMatrix()->MultTranspose(*dual,
+                                                                      *paraview_dual_grid_functions_[dual->name()]);
+        max_order_in_fields = std::max(max_order_in_fields, dual->space().GetOrder(0));
+        paraview_dc_->RegisterField(dual->name(), paraview_dual_grid_functions_[dual->name()].get());
+      }
+
```
Comment (btalamini) [inline src/serac/physics/base_physics.cpp]: Added code looks good. However, this whole block in the `if (!paraview_dc_)` block is getting long, and hence, while the logic in here is pretty simple, it's hard to follow at a glance. Consider folding all of this into a function like `CreateParaViewDataCollection` or something.

File: src/serac/physics/base_physics.cpp [hunk 2]
```diff
@@ -168,10 +187,18 @@ void BasePhysics::outputStateToDisk(std::optional<std::string> paraview_output_d
       for (const FiniteElementState* state : states_) {
         state->gridFunction();  // update grid function values
       }
+      for (const FiniteElementDual* dual : duals_) {
+        // These are really const calls, but MFEM doesn't label them as such
+        serac::FiniteElementDual* non_const_dual = const_cast<serac::FiniteElementDual*>(dual);
+        non_const_dual->space().GetRestrictionMatrix()->MultTranspose(*dual,
+                                                                      *paraview_dual_grid_functions_[dual->name()]);
+      }
```
Comment (btalamini) [inline src/serac/physics/base_physics.cpp]: It seems to me that the repetition of this code (from lines 156-161) could be avoided without too much effort. If I'm wrong, forget this comment, but otherwise, consider refactoring.

File: src/serac/physics/base_physics.cpp [hunk 3]
```diff
@@ -168,10 +187,18 @@ void BasePhysics::outputStateToDisk(std::optional<std::string> paraview_output_d
...
       for (auto& parameter : parameters_) {
         parameter.state->gridFunction();
       }
       shape_displacement_.gridFunction();
+      shape_displacement_sensitivity_->space().GetRestrictionMatrix()->MultTranspose(*shape_displacement_sensitivity_,
+                                                                                     *shape_sensitivity_grid_function_);
```
Comment (btalamini) [inline src/serac/physics/base_physics.cpp]: Same as previous comment.

**Review by btalamini — APPROVED** [review]
Looks good. I made some suggestions, but don't consider them prerequisite to merging.

File: src/serac/physics/base_physics.hpp [hunk 4]
```diff
@@ -534,6 +534,16 @@ class BasePhysics {
    */
   mutable std::unique_ptr<mfem::ParaViewDataCollection> paraview_dc_;
 
+  /**
+   * @brief A optional map of the dual names and duals in grid function form for paraview output
+   */
+  mutable std::unordered_map<std::string, std::unique_ptr<mfem::ParGridFunction>> paraview_dual_grid_functions_;
+
+  /**
+   * @brief A optional view of the shape sensitivity in grid funciton form for paraview output
```
Comment (samuelpmishLLNL) [inline src/serac/physics/base_physics.hpp]:
```suggestion
   * @brief A optional view of the shape sensitivity in grid function form for paraview output
```

**Review by samuelpmishLLNL — APPROVED** [review]
